sofa-ark-parent/core-impl/archive/src/main/java/com/alipay/sofa/ark/bootstrap/ClasspathLauncher.java

```diff
@@ -169,6 +183,108 @@ public InputStream getInputStream(ZipEntry zipEntry) throws IOException {
+            if (urls.length == 1) {
+                candidates = parseClassPathFromSurefireBoot(urls[0]);
+            } else {
+                candidates = urls;
+            }
+            URL[] filterUrls = filterURLs(candidates);
+            return filterUrls == null ? null : new DirectoryContainerArchive(filterUrls);
+        }
+
+        /**
+         * this method is used to chose jar file which is contained in sofa-ark-all.jar
+         *
+         * @return
+         */
+        protected URL[] filterURLs(URL[] urls) {
+            URL arkLibIndex = ClassLoader.getSystemResource("ark.lib.properties");
+
+            if (arkLibIndex == null) {
+                return null;
+            }
+
+            BufferedReader reader = null;
+            try {
+                reader = new BufferedReader(new InputStreamReader(arkLibIndex.openStream()));
+                Set<String> arkContainerJar = new HashSet<>();
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    arkContainerJar.add(line);
+                }
+
+                Set<URL> containerClassPath = new HashSet<>();
+                for (String arkJar : arkContainerJar) {
+                    for (URL url : urls) {
+                        if (url.getPath().contains(
+                            String.format(
+                                "%ssofa-ark-parent%score-impl%scontainer%starget%sclasses",
+                                File.separator, File.separator, File.separator, File.separator,
+                                File.separator))
+                            || url.getPath().contains(
+                                String.format(
+                                    "%ssofa-ark-parent%score-impl%sarchive%starget%sclasses",
+                                    File.separator, File.separator, File.separator, File.separator,
+                                    File.separator))
+                            || url.getPath().contains(
+                                String.format("%ssofa-ark-parent%score%sspi%starget%sclasses",
+                                    File.separator, File.separator, File.separator, File.separator,
+                                    File.separator))
+                            || url.getPath().contains(
+                                String.format("%ssofa-ark-parent%score%scommon%starget%sclasses",
+                                    File.separator, File.separator, File.separator, File.separator,
+                                    File.separator))
+                            || url.getPath().contains(
+                                String.format(
+                                    "%ssofa-ark-parent%score%sexception%starget%sclasses",
+                                    File.separator, File.separator, File.separator, File.separator,
+                                    File.separator))) {
+                            containerClassPath.add(url);
```

Comment:
These codes could be more elegant combined with  "ark.lib.properties" , currently it's a little bit hard to read them